### ISSUE TITLE
Latex Error Converting DVI problem

### ISSUE THREAD

**Issue by JackDusch** [3b1b/manim#610]
Include at least:
1. Steps to reproduce the issue (e.g. the command you ran)
2. The unexpected behavior that occurred (e.g. error messages or screenshots)
3. The environment (e.g. operating system and version of manim)

Hello, I have been choogling along with manim and am struggling like many to get the text functional. I managed to implement the changes suggested in #36 which is reiterated in several other threads. I am still able to run commands such as 

```
python3 ./manim.py example_scenes.py SquareToCircle -pl
```

However, I am experiencing an error message relating to the fix involving TEX_DIR that I am unsure how to fix. I tried pointing directly to the Tex directory within manim but didn't think that was consistent with the advice here either. My current tex_file_writing.py looks like:

```
def tex_to_dvi(tex_file):
    result = tex_file.replace(".tex", ".xdv")
    if not os.path.exists(result):
        commands = [
            "xelatex",
            "-no-pdf",
            "-interaction=batchmode",
            "-halt-on-error",
            "-output-directory=" TEX_DIR,
            tex_file,
            ">",
            get_null()
        ]
        exit_code = os.system(" ".join(commands))
        if exit_code != 0:
            log_file = tex_file.replace(".tex", ".log")
            raise Exception(
                "Latex error converting to dvi."
                "See log output above or the log file: %s" % log_file)
    return result   


def dvi_to_svg(dvi_file, regen_if_exists=False):
    """
    Converts a dvi, which potentially has multiple slides, into a
    directory full of enumerated pngs corresponding with these slides.
    Returns a list of PIL Image objects for these images sorted as they
    where in the dvi
    """
    result = dvi_file.replace(".xdv", ".svg")
    if not os.path.exists(result):
        commands = [
            "dvisvgm",
            dvi_file,
            "-n",
            "-v",
            "0",
            "-o",
            result,
            ">",
            get_null()
        ]
        os.system(" ".join(commands))
    return result
```

But, when I attempt to run 
```
python3 ./manim.py example_scenes.py Write Stuff -pl
```
as a test, 

![Screen Shot 2019-06-28 at 10 55 19 PM](https://user-images.githubusercontent.com/51562821/60378926-387a7780-99f8-11e9-94fe-fdf96510e2ad.png)

This is all I can muster. I can still run code not involving text. I tried messing around and removing the TEX_DIR portion, which yields the following message:

![Screen Shot 2019-06-28 at 10 59 49 PM](https://user-images.githubusercontent.com/51562821/60378953-a626a380-99f8-11e9-861f-3a7c3d77a4b0.png)

I can't wait to share my work if we can get through this! I appreciate everyone's consideration.

**Comment by Kolloom** [comment]
Not sure what version you are using. 
It should be https://github.com/3b1b/manim/blob/master/manimlib/utils/tex_file_writing.py#L47

**Comment by JackDusch** [comment]
So I am using OS X and it should be the latest version. On blog posts and throughout these issues I see mostly people on OS X using the format of text_to_dvi and dvi_to_svg that I posted. 
So I should be changing back? There were some issues using dvi on OS X I thought I understood these changes were working around the issues, as in #300, #36. 

Additionally, despite error messages pointing to logs in Output or Tex I find both directories are empty. During installation I did set the proper Output according to 
http://bhowell4.com/manic-install-tutorial-for-mac/

This blog post guided me through installation, and setting the output, then it instructs me to make the changes above for fixing Latex on OS X. I have the latest MacTex installed as of two days ago. 

If anyone has any ideas based on my info and the steps in the blog post I would really appreciate it.

**Comment by Kolloom** [comment]
the two issues you linked are due to missing latex packages, which is not what your error output suggests.

**Comment by JackDusch** [comment]
Yes that is true, these were the issues I was experiencing that led me to this point, I'm just looking for guidance on what to do from here. Appreciated!

**Comment by JackDusch** [comment]
With the form you provided, the output changes to this: 
![Screen Shot 2019-06-30 at 5 57 28 PM](https://user-images.githubusercontent.com/51562821/60402660-a477ef80-9b60-11e9-81cf-d3549bb73b34.png)

**Comment by Kolloom** [comment]
Modifying the code without understanding how they fit together would not likely yield much result. 

The article is not dated. Judging from the comment section its from December last year. There has been a few changes to the file writing so the modification from the article should no longer be needed. 
Please give this a try: https://github.com/3b1b/manim/blob/master/docs/source/installation/mac.rst

**Comment by yoshiask** [comment]
Before you replaced tex_file_writing.py, the error message was
`AttributeError: module 'manimlib.constants' has no attribute 'TEXT_DIR'`.

Line 48 in tex_to_dvi should have `consts.TEX_DIR`, not `consts.TEXT_DIR`. If you can, revert back to the form that @Kolloom suggested and change line 48.

**Comment by AamodJ** [comment]
> So I am using OS X and it should be the latest version. On blog posts and throughout these issues I see mostly people on OS X using the format of text_to_dvi and dvi_to_svg that I posted.
> So I should be changing back? There were some issues using dvi on OS X I thought I understood these changes were working around the issues, as in #300, #36.
> 
> Additionally, despite error messages pointing to logs in Output or Tex I find both directories are empty. During installation I did set the proper Output according to
> http://bhowell4.com/manic-install-tutorial-for-mac/
> 
> This blog post guided me through installation, and setting the output, then it instructs me to make the changes above for fixing Latex on OS X. I have the latest MacTex installed as of two days ago.
> 
> If anyone has any ideas based on my info and the steps in the blog post I would really appreciate it.

I'm facing the second part of your issue. 
I am on OS X, and followed bhowell's blogpost for getting things working. Back in december, I could use TextMobjects. But later on after formatting my drive (for reasons) and reinstalling manim and all it's dependencies, it's giving me the error that get_null is not defined. It'll be of great help if someone could help

**Comment by AamodJ** [comment]
I am running manim on OS X and I just cloned the repo and everything works fine (if you have all the dependencies right).
I think you might have a wrong version of LaTeX. Did you install MikTeX or MacTeX? I've installed MacTeX by ```brew cask install mactex```. If you don't have MacTeX, try installing and trying again. And also, revert your ```tex_file_writing.py``` file to the latest one from https://github.com/3b1b/manim/blob/master/manimlib/utils/tex_file_writing.py

**Comment by girishgargcool** [comment]
i am having same problem. please help at #623
